tests/adapter/dbt/tests/adapter/test_incremental_unique_id.py

```diff
@@ -0,0 +1,597 @@
+        row_count_query = "select * from {}.{}".format(project.test_schema, seed)
+        project.run_sql_file(Path("seeds") / Path(update_sql_file + ".sql"))
+        seed_rows = len(project.run_sql(row_count_query, fetch="all"))
+
+        # propagate seed state to incremental model according to unique keys
+        inc_test_model_count = self.update_incremental_model(incremental_model=incremental_model)
+
+        return (seed_count, model_count, seed_rows, inc_test_model_count)
+
+    def check_scenario_correctness(self, expected_fields, test_case_fields, project):
+        """Invoke assertions to verify correct build functionality"""
+        # 1. test seed(s) should build afresh
+        assert expected_fields.seed_count == test_case_fields.seed_count
+        # 2. test model(s) should build afresh
+        assert expected_fields.model_count == test_case_fields.model_count
+        # 3. seeds should have intended row counts post update
+        assert expected_fields.seed_rows == test_case_fields.seed_rows
+        # 4. incremental test model(s) should be updated
+        assert expected_fields.inc_test_model_count == test_case_fields.inc_test_model_count
+        # 5. extra incremental model(s) should be built; optional since
+        #   comparison may be between an incremental model and seed
+        if expected_fields.opt_model_count and test_case_fields.opt_model_count:
+            assert expected_fields.opt_model_count == test_case_fields.opt_model_count
+        # 6. result table should match intended result set (itself a relation)
+        table_comp = TableComparison(
+            adapter=project.adapter, unique_schema=project.test_schema, database=project.database
+        )
+        table_comp.assert_tables_equal(expected_fields.relation, test_case_fields.relation)
+
+    def stub_expected_fields(self, relation, seed_rows, opt_model_count=None):
+        return ResultHolder(
+            seed_count=1,
+            model_count=1,
+            seed_rows=seed_rows,
+            inc_test_model_count=1,
+            opt_model_count=opt_model_count,
+            relation=relation,
+        )
+
+    def fail_to_build_inc_missing_unique_key_column(self, incremental_model_name):
+        """should pass back error state when trying build an incremental
+        model whose unique key or keylist includes a column missing
+        from the incremental model"""
+        # seed_count = len(run_dbt(["seed", "--select", "seed", "--full-refresh"]))  # noqa:F841
+        # unique keys are not applied on first run, so two are needed
+        run_dbt(
+            ["run", "--select", incremental_model_name, "--full-refresh"],
+            expect_pass=True,
+        )
+        run_result = run_dbt(
+            ["run", "--select", incremental_model_name], expect_pass=False
+        ).results[0]
+
+        return run_result.status, run_result.message
+
+
+class TestIncrementalWithoutUniqueKey(IncrementalUniqueKeyBase):
```

Comment:
None of these tests are actually importable, because they start with the name 'Test'.  Why aren't the tests in the base test class?

Reply:
the original tests were separated out into classes that inherited the setup based on type of unique_key case it was testing. do we have a preferred way to reformat the test/class names?

Reply:
moved all tests to the base class and created a new test class based on the other examples we have.

Reply:
Ah, this conversation clarified for me why we're doing it like this:

https://github.com/dbt-labs/dbt-core/blob/5071b00baac3ff2a9bcce565d09661605de52654/tests/adapter/dbt/tests/adapter/basic/test_base.py#L109-L110

As I understand it:
- By default, `pytest` only runs tests if the method starts with `test_`, and the class starts with `Test`, in a file whose name starts with `test_`
- Our move has been to create a "base" class (_not_ named `Test`), with test methods defined on it that _do_ start with `test_*`
- Then, inherit that "base" class into an actual test class, whose name starts with `Test`, and either pass (no changes needed) or override/reimplement fixtures/methods from the base class

Reply:
Yes, that's correct :-). You *can* import a base class that starts with Test, the problem is that pytest would then run all of the tests in the base class plus all of those same tests in the subclass. Which means that if you need to override a test method or get failures, you'd get failures.